File: 49-GroupAnagrams/answer.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        wordToAn = {}
        for word in strs:
            sWord = tuple(sorted(word))
            if sWord in wordToAn:
                wordToAn[sWord].append(word)
            else:
                wordToAn[sWord] = [word]

        return wordToAn.values()

File: 49-GroupAnagrams/answer.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();

        for (int i = 0; i < strs.length; i++){

            char sortedArr[] = strs[i].toCharArray();
            Arrays.sort(sortedArr);
            String sortedWord = new String(sortedArr);

            if (!map.containsKey(sortedWord)){
                map.put(sortedWord, new ArrayList<>());
            }

            map.get(sortedWord).add(strs[i]);
        }

        return new ArrayList<>(map.values());
    }

}